static/js/brand-store/shared/PasswordToggle/PasswordToggle.test.js

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import "@testing-library/jest-dom";
+
+import PasswordToggle from "./PasswordToggle";
+
+const testProps = {
+  id: "foo",
+  label: "Password",
+  value: "asdfasdfsdafsda",
+  readOnly: false,
+};
+
+test("it renders the label", () => {
+  render(<PasswordToggle {...testProps} />);
+  expect(screen.getByText(testProps.label)).toBeInTheDocument();
+});
+
+test("it has the correct value in the password field", () => {
+  render(<PasswordToggle {...testProps} />);
+  expect(screen.getByLabelText(testProps.label).value).toEqual(testProps.value);
+});
+
+test("it shows the correct button text when password is hidden", () => {
+  render(<PasswordToggle {...testProps} />);
+  expect(screen.getByTestId("show-button")).toBeInTheDocument();
+  expect(screen.queryByTestId("hide-button")).not.toBeInTheDocument();
```

Comment:
This test isn't really checking the button text, but the implementation detail of the test id. You might want to select the button by it's role and then use [toHaveTextContent](https://github.com/testing-library/jest-dom#tohavetextcontent) on it.